### PR TITLE
chore: iOS toolchain update

### LINKED ISSUE #1796: [iOS] Fix build hints and warnings
The iOS build currently has a number of hints and warnings. We should resolve these before they grow further:

```
[11:01:43][Step 2/2] === BUILD TARGET KeymanEngine OF PROJECT KeymanEngine WITH CONFIGURATION Release ===
[11:01:43][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/SubKeysView.swift:89:9: warning: variable 'viewHeight' was never mutated; consider changing to 'let' constant
[11:01:43][Step 2/2]     var viewHeight = baseHeight + containerHeight - marginY + strokeWidth
[11:01:43][Step 2/2]     ~~~ ^
[11:01:43][Step 2/2]     let
[11:01:43][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/SubKeysView.swift:91:9: warning: variable 'viewPosY' was never mutated; consider changing to 'let' constant
[11:01:43][Step 2/2]     var viewPosY = keyFrame.origin.y - (viewHeight - keyFrame.size.height)
[11:01:43][Step 2/2]     ~~~ ^
[11:01:43][Step 2/2]     let
[11:01:43][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/Manager.swift:643:17: warning: variable 'languageName' was written to, but never read
[11:01:43][Step 2/2]             var languageName = ""
[11:01:43][Step 2/2]                 ^
[11:01:43][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/KeyPreviewView.swift:33:9: warning: variable 'viewHeight' was never mutated; consider changing to 'let' constant
[11:01:43][Step 2/2]     var viewHeight = keyFrame.size.height * 2.1
[11:01:43][Step 2/2]     ~~~ ^
[11:01:43][Step 2/2]     let
[11:01:43][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/KeyPreviewView.swift:36:9: warning: variable 'viewPosY' was never mutated; consider changing to 'let' constant
[11:01:43][Step 2/2]     var viewPosY = keyFrame.origin.y - (viewHeight - keyFrame.height)
[11:01:43][Step 2/2]     ~~~ ^
[11:01:43][Step 2/2]     let
[11:01:43][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/KeyPreviewView.swift:38:9: warning: initialization of immutable value 'tbHeight' was never used; consider replacing with assignment to '_' or removing it
[11:01:43][Step 2/2]     let tbHeight = Manager.shared.inputViewController.activeTopBarHeight
[11:01:43][Step 2/2]     ~~~~^~~~~~~~
[11:01:43][Step 2/2]     _
[11:01:43][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/Model/KeymanPackage.swift:61:18: warning: value 'packagedKeyboards' was defined but never used; consider replacing with boolean test
[11:01:43][Step 2/2]           if let packagedKeyboards = jsonResult["keyboards"] as? [[String:AnyObject]] {
[11:01:43][Step 2/2]              ~~~~^~~~~~~~~~~~~~~~~~~~
[11:01:43][Step 2/2]              (                                                                       ) != nil
[11:01:43][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/Model/KeymanPackage.swift:62:20: warning: value 'packagedModels' was defined but never used; consider replacing with boolean test
[11:01:43][Step 2/2]             if let packagedModels = jsonResult["lexicalModels"] as? [[String:AnyObject]] {
[11:01:43][Step 2/2]                ~~~~^~~~~~~~~~~~~~~~~
[11:01:43][Step 2/2]                (                                                                        ) != nil
[11:01:43][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/Model/KeymanPackage.swift:71:23: warning: value 'packagedModels' was defined but never used; consider replacing with boolean test
[11:01:43][Step 2/2]           else if let packagedModels = jsonResult["lexicalModels"] as? [[String:AnyObject]] {
[11:01:43][Step 2/2]                   ~~~~^~~~~~~~~~~~~~~~~
[11:01:43][Step 2/2]                   (                                                                        ) != nil
[11:01:44][Step 2/2] === BUILD TARGET KeymanEngine OF PROJECT KeymanEngine WITH CONFIGURATION Release ===
[11:01:44][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/SubKeysView.swift:89:9: warning: variable 'viewHeight' was never mutated; consider changing to 'let' constant
[11:01:44][Step 2/2]     var viewHeight = baseHeight + containerHeight - marginY + strokeWidth
[11:01:44][Step 2/2]     ~~~ ^
[11:01:44][Step 2/2]     let
[11:01:44][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/SubKeysView.swift:91:9: warning: variable 'viewPosY' was never mutated; consider changing to 'let' constant
[11:01:44][Step 2/2]     var viewPosY = keyFrame.origin.y - (viewHeight - keyFrame.size.height)
[11:01:44][Step 2/2]     ~~~ ^
[11:01:44][Step 2/2]     let
[11:01:44][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/Manager.swift:643:17: warning: variable 'languageName' was written to, but never read
[11:01:44][Step 2/2]             var languageName = ""
[11:01:44][Step 2/2]                 ^
[11:01:44][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/KeyPreviewView.swift:33:9: warning: variable 'viewHeight' was never mutated; consider changing to 'let' constant
[11:01:44][Step 2/2]     var viewHeight = keyFrame.size.height * 2.1
[11:01:44][Step 2/2]     ~~~ ^
[11:01:44][Step 2/2]     let
[11:01:44][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/KeyPreviewView.swift:36:9: warning: variable 'viewPosY' was never mutated; consider changing to 'let' constant
[11:01:44][Step 2/2]     var viewPosY = keyFrame.origin.y - (viewHeight - keyFrame.height)
[11:01:44][Step 2/2]     ~~~ ^
[11:01:44][Step 2/2]     let
[11:01:44][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/KeyPreviewView.swift:38:9: warning: initialization of immutable value 'tbHeight' was never used; consider replacing with assignment to '_' or removing it
[11:01:44][Step 2/2]     let tbHeight = Manager.shared.inputViewController.activeTopBarHeight
[11:01:44][Step 2/2]     ~~~~^~~~~~~~
[11:01:44][Step 2/2]     _
[11:01:44][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/Model/KeymanPackage.swift:61:18: warning: value 'packagedKeyboards' was defined but never used; consider replacing with boolean test
[11:01:44][Step 2/2]           if let packagedKeyboards = jsonResult["keyboards"] as? [[String:AnyObject]] {
[11:01:44][Step 2/2]              ~~~~^~~~~~~~~~~~~~~~~~~~
[11:01:44][Step 2/2]              (                                                                       ) != nil
[11:01:44][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/Model/KeymanPackage.swift:62:20: warning: value 'packagedModels' was defined but never used; consider replacing with boolean test
[11:01:44][Step 2/2]             if let packagedModels = jsonResult["lexicalModels"] as? [[String:AnyObject]] {
[11:01:44][Step 2/2]                ~~~~^~~~~~~~~~~~~~~~~
[11:01:44][Step 2/2]                (                                                                        ) != nil
[11:01:44][Step 2/2] /Users/bob/teamcity/work/fce8cb6628d21d3f/keyman/ios/engine/KMEI/KeymanEngine/Classes/Model/KeymanPackage.swift:71:23: warning: value 'packagedModels' was defined but never used; consider replacing with boolean test
[11:01:44][Step 2/2]           else if let packagedModels = jsonResult["lexicalModels"] as? [[String:AnyObject]] {
[11:01:44][Step 2/2]                   ~~~~^~~~~~~~~~~~~~~~~
[11:01:44][Step 2/2]                   (                                                                        ) != nil
```

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #1796.  Well, given its nature, it's more of an "update", really... but the goal is to remove all build warnings.

Now that macOS is up to 10.15 (with corresponding Xcode updates) and iOS is well into version 13.x, we should probably update the iOS codebase accordingly.  Along with this, we may as well update to Swift 5.1 - there are some serious benefits for doing so:

- https://swift.org/blog/abi-stability-and-more/
- https://medium.com/swiftify/swift-5-1-module-format-stability-best-time-migrate-objective-c-frameworks-a0434f5352a3

The tl;dr of these links - Apple finally cares about backward compatibility for Swift dependencies.  Once we update to use of 5.1, concern about exactly which compiler was used for our published library (KeymanEngine.framework) should go away.

----

As part of the toolchain update, Apple's deprecated the old raw-image launch style in favor of "storyboard"-based launch screens.  This is now incorporated as part of this PR - see comments below.

### CODE REVIEW

**Comment by jahorton** [comment]
Build agent failure:  the build is failing on codesigning... when attempting to avoid it.  That'll take a bit of work, too.

**Comment by mcdurdin** [comment]
So far so good ... but gee Apple, move all the deckchairs why don't you?

**Comment by jahorton** [comment]
The launch "splash" screens are now generated via storyboard:

light mode launch

iPhone SE light launch

This is done via two "views" - one for the Keyman logo and one for the color splash bar at the bottom.  As seen here, both scale and reposition appropriately.  (Even in landscape!)

For a bonus, now that dark mode is a thing on iOS:

dark mode launch

It took a bit of image editing on the logo to clean up the inverted text, but the new launch storyboard is ready for dark mode.  (As mentioned by https://www.fivestars.blog/code/ios-dark-mode-how-to.html, compiling with Xcode 11 automatically enables dark mode!)

That stated... this can't be said for the rest of the app:

naive dark mode fail

Ignoring the predictive banner issue, note the "dark mode" nav bar at the top.  Making that look right will take a bit of work.  The "Get Started" view also has some nasty dark-mode styling issues.  For now, I've added a commit that disables dark mode entirely - we can fix that up at a later point.  (This would be a reasonable point of divergence for a cherrypick against stable, I think; it's probably best to keep 12.0 stable in "light-only" mode for now.)

----

Now, back to the predictive banner issue - it's actually not just the banner, but is something affecting the whole of KMW.  Debugging with Safari indicates that iPads in Simulator are being recognized as "macOS" in KMW within the app.  This likely has to do with some significant developments with Xcode 11 and macOS 10.15 in relation to the new Catalyst feature on macOS.  I'm not 100% sure yet, but I'm suspecting that similar to dark mode, this may be something related to compiling with Xcode 11.  It's hard to say without having access to an iOS 13.x physical test device, though.

Either way, it's quite important that we address this final detail (or prove it a Simulator-only bug) before publishing this PR.

**Comment by jahorton** [comment]
The Safari debugging session indicating the "macOS" state in Simulator:

safari debug

"Get Started"'s current dark mode state:

dark mode get started fail

"Get Started" and the app's base view navigation bar seem to be the only two locations that are currently adversely affected by dark mode.  ("Text size" could also use a mild restyle, but is at least internally consistent and coherently styled.)  It would probably be possible to simply "light-mode" style the nav bar and fix the other issue without too much effort, giving us a basic level of Dark Mode support immediately if desired.

**Comment by jahorton** [comment]
For tests on the current state of this PR:

If it's possible to delay this step, doing so for testing may be useful/revealing:
- Update host (building) machine to macOS 10.15
  - There's a chance the version of host macOS is related.

The rest of the steps are necessary:
- Update your development machine to Xcode 11
- `git checkout ios-toolchain-update`
- `./build.sh -debug`
- From Xcode, run and test on tablet form factors.

If you want to "throw in" some dark-mode testing:

(Simpler approach)
- In the iOS Xcode workspace, go to Keyman > Keyman > Supporting Files > Keyman-Info.plist
- Find the "User Interface Style" entry and set its value to "Dark"
  - If later changes to this PR remove the entry, simply re-add it.  "Light" was the original value, which forced "light" mode.

("Correct", but more tedious approach)
- (Alternative) Remove the entry above and use iOS Settings > Developer > Dark Appearance to control it 'natively'.

**Comment by jahorton** [comment]
Found some useful info regarding the macOS Simulator - it won't be just the Simulator:

https://forums.developer.apple.com/thread/119186#373517

Recent Apple changes, along with the "iPadOS", mean that Catalyst and iPad will have matching user agent strings, at least for now.  They've even changed the direct-linked workaround; the user agent string will still report Mac OS X even when not requesting desktop pages.  We'll have to add extra code for detecting / differentiating as a result.

One possible angle:  https://stackoverflow.com/questions/56578799/tell-ipados-from-macos-on-the-web

For embedded mode, we can use `util.device.app` to override the user agent string, but that won't help KMW "native" on iPads.

**Comment by jahorton** [comment]
The issue above has been isolated and identified as #2267 and will be addressed separately in #2268.

**Comment by jahorton** [comment]
- [ ] Ah, there is one more detail - the default text color shifts to white in dark mode.

**Comment by jahorton** [comment]
I've spun the basic dark mode implementation into its own PR - #2277.

**Comment by mcdurdin** [comment]
I've started this process -- 16GB of updates required on my Mac. Ugh. Will continue once it finishes (it should go overnight... can't believe that I still have to schedule overnight downloads for macOS).

**Comment by jahorton** [comment]
Right.  I've run tests by creating a temporary testing branch where I merge the UA-spoof branch `web-2267-detect-ios-spoofs` with this branch `ios-toolchain-update` when I want to see the complete picture.

That said, this PR aims to make minimal actual changes to the code; it serves as a fix for deprecations and toolchain issues moving forward.  

I'd say that the biggest point of worry for testing is with the toolbar options.  Some of them create small overlays, and there were a far amount of changes to that code, as the old "popover" implementation was deprecated; it got changed from its own UI element type to a style that can be applied to other elements.

Most important file:
* ios/keyman/Keyman/Keyman/MainViewController.swift

Past that, just launch it under multiple Simulator configurations (different devices) and see if the launch screen "looks right", as that's the other "big change."  Granted, you can visually test it in the storyboard editor within Xcode, so I'm not too worried on that point.

To test:
* Go to ios/keyman/Keyman/LaunchScreen.storyboard within Xcode
* You can see how everything is laid out for different devices and even examine dark vs light mode behavior there.

**Comment by mcdurdin** [comment]
OK, I am comfortable with merging for now -- it seems right, in use.
